geopy/geocoders/here.py

```diff
@@ -0,0 +1,281 @@
+
+        :rtype: ``None``, :class:`geopy.location.Location` or a list of them, if
+            ``exactly_one=False``.
+        """
+        if isinstance(query, dict):
+            params = {
+                key: val
+                for key, val
+                in query.items()
+                if key in self.structured_query_params
+            }
+            params['app_id'] = self.app_id
+            params['app_code'] = self.app_code
+        else:
+            params = {
+                'searchtext': self.format_string % query,
+                'app_id': self.app_id,
+                'app_code': self.app_code
+            }
+        if exactly_one:
+            params['maxresults'] = 1
+        if language:
+            params['language'] = language
+        if additional_data:
+            params['additionaldata'] = additional_data
+        if other:
+            params.update(other)
+
+        url = "?".join((self.api, urlencode(params)))
+        logger.debug("%s.geocode: %s", self.__class__.__name__, url)
+        return self._parse_json(
+            self._call_geocoder(url, timeout=timeout),
+            exactly_one
+        )
+
+    def reverse(
+            self,
+            query,
+            exactly_one=True,
+            language=None,
+            other=None,
+            timeout=DEFAULT_SENTINEL
+    ):
+        """
+        Return an address by location point.
+
+        Documentation at:
+            https://developer.here.com/documentation/geocoder/topics/resource-reverse-geocode.html
+
+        :param query: The coordinates for which you wish to obtain the
+            closest human-readable addresses.
+        :type query: :class:`geopy.point.Point`, list or tuple of ``(latitude,
+            longitude)``, or string as ``"%(latitude)s, %(longitude)s"``.
+
+        :param bool exactly_one: Return one result or a list of results, if
+            available.
+
+        :param int timeout: Time, in seconds, to wait for the geocoding service
+            to respond before raising a :class:`geopy.exc.GeocoderTimedOut`
+            exception. Set this only if you wish to override, on this call
+            only, the value set during the geocoder's initialization.
+
+        :param str language: Affects the language of the response,
+            must be a RFC 4647 language code, e.g. 'en-US'.
+
+        :param dict other: A dictionary with additional key/values as described
+            in the online documentation. These will be added as individual query
+            parameters to the URL.
+
+        :rtype: ``None``, :class:`geopy.location.Location` or a list of them, if
+            ``exactly_one=False``.
+        """
+        point = self._coerce_point_to_string(query)
+        params = {
+            'app_id': self.app_id,
+            'app_code': self.app_code,
+            'maxresults': 1,
+            'mode': 'retrieveAddresses',  # can be overwritten using `other`
+            'prox': point + ',1000',
```

Comment:
Is radius value required? If not, it would be better to not specify it explicitly. But in any way it should be a separate kwarg, so it could be changed.

Reply:
Ok, will test and make a separate param if possible.